File: gateway/gateway-api/src/main/java/ru/gilko/gatewayapi/dto/statistic/CancelingStatisticDto.java
package ru.gilko.gatewayapi.dto.statistic;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class CancelingStatisticDto {
    private ProfitableModelDto profitableModel;
    private int day;
    private int daysToEnd;
}

File: frontend/src/dtos/CancelingStatisticDto.js
import type ProfitableModelDto from "./ProfitableModelDto";

class CancelingStatisticDto {
    profitableModel: ProfitableModelDto;
    day: number;
    daysToEnd: number;

    constructor(profitableModel: ProfitableModelDto,
                day: number,
                daysToEnd: number) {
        this.profitableModel = profitableModel;
        this.day = day;
        this.daysToEnd = daysToEnd;
    }
}

export default CancelingStatisticDto;